runner/runner.go

```diff
@@ -140,6 +140,56 @@ func (r *Runner) Run(ctx context.Context, args []string) int {
 	// application & backend setup should be done before starting any new goroutines.
 	r.application.Setup()
 
+	// START - OpenTelemetry setup
+	var (
+		statsEnabled        = config.GetBool("enableStats", true)
+		otelTracesEndpoint  = config.GetString("OpenTelemetry.Traces.Endpoint", "")
+		otelMetricsEndpoint = config.GetString("OpenTelemetry.Metrics.Endpoint", "")
+	)
+	if statsEnabled && otelTracesEndpoint == "" && otelMetricsEndpoint == "" {
+		r.logger.Warnf("No OpenTelemetry endpoints provided")
+	} else if statsEnabled {
+		instanceID := config.GetString("INSTANCE_ID", "")
+		res, err := otel.NewResource(r.appType, instanceID, r.releaseInfo.Version,
+			attribute.String("instanceName", instanceID),
+			attribute.String("namespace", config.GetNamespaceIdentifier()),
+		)
+		if err != nil {
+			r.logger.Errorf("OpenTelemetry resource creation failed: %v", err)
+			return 1
+		}
+
+		var (
+			otelManager otel.Manager
+			otelOpts    = []otel.Option{otel.WithInsecureGRPC(), otel.WithLogger(r.logger.Child("otel"))}
+		)
+		if otelTracesEndpoint != "" {
+			otelOpts = append(otelOpts, otel.WithTracerProvider(otelTracesEndpoint, otel.WithGlobalTracerProvider()))
+		}
+		if otelMetricsEndpoint != "" {
+			otelOpts = append(otelOpts, otel.WithMeterProvider(otelMetricsEndpoint,
+				otel.WithGlobalMeterProvider(),
+				otel.WithMeterProviderExportsInterval(
+					config.GetDuration("OpenTelemetry.Metrics.ExportInterval", 5, time.Second),
+				),
+			))
+		}
+		if _, _, err = otelManager.Setup(ctx, res, otelOpts...); err != nil {
+			r.logger.Errorf("OpenTelemetry setup failed: %v", err)
+			return 1
+		}
```

Comment:
Yes that is correct. That is a blocking call that retries a few times and returns an error if a connection cannot be established. I can make it optional if preferred and have RS start anyway. To avoid bloating the server memory I'll have to skip everything until we can connect.